server/plugins/messageStorage/sqlite.ts

```diff
@@ -44,18 +44,36 @@ class Deferred {
 	}
 }
 
+type SqliteConfig = typeof Config.values.sqliteConfig;
+
 class SqliteMessageStorage implements SearchableMessageStorage {
 	isEnabled: boolean;
 	database!: Database;
 	initDone: Deferred;
 	userName: string;
+	mergedConfig: typeof Config.values.sqliteConfig;
 
 	scheduledIntervalId: ReturnType<typeof setInterval> | undefined;
 
-	constructor(userName: string) {
+	constructor(userName: string, clientStorageConfig: SqliteConfig) {
 		this.userName = userName;
 		this.isEnabled = false;
 		this.initDone = new Deferred();
+
+		this.mergedConfig = this.mergeClientConfig(clientStorageConfig);
+	}
+
+	mergeClientConfig(clientStorageConfig: SqliteConfig) {
```

Comment:
This isn't the right place for that kind of logic.
The config needs to be merged by the core code, not the sqlite specific storage backend.

Yes, currently sqlite is our only implementation of the interface, but that doesn't mean that we should violate the layers.